### PR TITLE
[OPS-889] Check .cabal files into the repository

### PR DESCRIPTION
**Pull request by zhenyavinogradov**
Fixes using this library as a git dependency in other cabal and stack>2
projects

### CODE REVIEW

File: shell.nix [hunk 1]
```diff
@@ -1,7 +1,6 @@
 with import <nixpkgs> { };
-let hspkgs = haskell.packages.ghc822; in
 haskell.lib.buildStackProject {
-   ghc = hspkgs.ghc;
+   ghc = haskell.compiler.ghc822Binary;
```
Comment (kirelagin) [inline shell.nix]: oh no, let’s wait for #50.

**Comment by kirelagin** [comment]
@zhenyavinogradov #50 has been merged, pls update this to current master.

**Comment by yorickvP** [comment]
stack > 2 supports hpack, doesn't it?

**Comment by zhenyavinogradov** [comment]
Stack 2 doesn't work for git dependencies with package.yaml files referencing the parent directory, like in this repository

**Comment by yorickvP** [comment]
That sounds like a stack bug

**Comment by zhenyavinogradov** [comment]
It's a known problem, but stack is keeping it as it is: https://github.com/commercialhaskell/stack/issues/4906

**Comment by yorickvP** [comment]
Okay, so unsharing the yaml would also work? I'm not sure if checking build outputs into git is good if we can avoid it.

**Comment by kirelagin** [comment]
Cabal file is not a build output.

**Comment by zhenyavinogradov** [comment]
Unsharing package.yaml files would mostly kill their purpose of not having to repeat the same information in .cabal files. So providing .cabal file is the only proper way to make a library usable as a dependency in cabal and stack>=2 projects. As an option, we can provide an archive with .cabal files included in a separate place, like hackage or github releases.

**Pull request merged**